packages/kbn-content-management-utils/src/saved_object_content_storage.ts

```diff
@@ -280,13 +290,28 @@ export abstract class SOContentStorage<Types extends CMCrudTypes>
       createOptions
     );
 
+    const result = {
+      item: savedObjectToItem(savedObject, this.allowedSavedObjectAttributes, false),
+    };
+
+    const validationError = transforms.create.out.result.validate(result);
+    if (validationError) {
+      if (this.throwOnResultValidationError) {
+        throw Boom.badRequest(`Invalid response. ${validationError.message}`);
+      } else {
+        this.logger.warn(`Invalid response. ${validationError.message}`);
+      }
+    }
+
     // Validate DB response and DOWN transform to the request version
     const { value, error: resultError } = transforms.create.out.result.down<
       Types['CreateOut'],
       Types['CreateOut']
-    >({
-      item: savedObjectToItem(savedObject, this.allowedSavedObjectAttributes, false),
-    });
+    >(
```

Comment:
Another option would be to update the `down` function and always return the `value` even if there is an error?

Reply:
As you mentioned it is more work, need packages and tests changes, etc, but I am more worried about types as we will have to change them also. Now `down` accepts `DownIn` and returns `DownOut`, but looks like it will have to return `DownIn | DownUp`

Reply:
Wouldn't it be simply that: if error is return, `DownIn` is returned and if error is `null` or `undefined` then `DownOut` is returned?

Reply:
I kept the initial approach because the result would have been isn't the same. 
In the current approach we could fail validation and log a warning, but we still would attempt to run migrations. if those won't fail - we respond, if those fail, we would error